gtsam/discrete/DiscreteConditional.cpp

```diff
@@ -135,7 +135,7 @@ DecisionTreeFactor::shared_ptr DiscreteConditional::choose(
   // Convert ADT to factor.
   DiscreteKeys discreteKeys;
   for (Key j : frontals()) {
-    discreteKeys.emplace_back(j, this->cardinality(j));
+    discreteKeys.emplace(j, this->cardinality(j));
```

Comment:
Actually, this has to stay a vector I think, because the order of parents matters !